### PR TITLE
Implement TextPositionSelector, create Chunking abstraction

### PR DESCRIPTION
**Pull request by Treora**
Finally.

- Addresses #75 (support TextPositionSelector)
- Addresses #85 (Chunking abstraction); because having two selectors dealing with text was the right moment to introduce such abstractions. Now the logic of matching&describing a selector is in the selector package, and operates on Chunks (an arbitrary object wrapping a string). Hence the dom package’s main task now is to merely convert a Range into a sequence of chunks (one chunk per text node), and call the abstract implementation with that.
- Reintroduces Range normalisation (see #51), implemented from scratch.
- Reimplements and remove need for dom-seek dependency.
- Updates the demo to show TextPosition, adding a smiley (= two code units) and some elements to make it more challenging.

Things to do still (can wait until after merging):
- Document everything better; especially the details of the chunking approach.
- Make tests for the abstract match&describe implementations; test the abstraction layers individually (TextNodeChunker, TextSeeker, CodePointSeeker, normalizeRange); decide how to test the dom-specific match&describe implementations.
- Possibly also document&publish the abstractions (e.g. CodePointSeeker) as separate components that could be helpful for other purposes than annotation.
- Modify TextQuoteMatcher to use Seeker, not Chunker (I built further on the reimplementation I made in the chunking branch)

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -111,6 +112,7 @@ module.exports = {
       plugins: ['@typescript-eslint'],
       rules: {
         '@typescript-eslint/consistent-type-imports': 'error',
+        '@typescript-eslint/no-explicit-any': 'off',
```
Comment (tilgovi) [inline .eslintrc.js:115]: I agree. We should avoid it where possible but sometimes libraries need to be permissive or ignorant about the types that are passed through APIs.

File: packages/dom/src/normalize-range.ts [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+// of methods that would modify these nodes (note that a user can simply cast
+// the TextRange back to a Range to remove these restrictions).
+export interface TextRange extends Range {
+  readonly startContainer: Text;
+  readonly endContainer: Text;
+  cloneRange(): TextRange;
+
+  // Allow only Text nodes to be passed to these methods.
+  insertNode(node: Text): void;
+  selectNodeContents(node: Text): void;
```
Comment (tilgovi) [inline packages/dom/src/normalize-range.ts:34]: What should we do with `selectNode` (as opposed to `selectNodeContents`).

Comment (tilgovi) [reply packages/dom/src/normalize-range.ts:34]: Oh, I see. It's defined below with `never`.

**Review by tilgovi — APPROVED** [review]
I'm not :100: on all the APIs yet, but I'd like to merge and then make incremental changes from there. Excellent stuff.

**Comment by Treora** [comment]
> I'm not 100 on all the APIs yet, …

Me neither.

> …but I'd like to merge and then make incremental changes from there. Excellent stuff.

Cool with me! I squashed some of the commits to make the path towards this a little more comprehensible.

**Pull request merged**